presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewInformationExtractor.java

```diff
@@ -173,5 +168,26 @@ public boolean isDistinct()
         {
             return isDistinct;
         }
+
+        public Optional<Set<Expression>> getGroupBy()
+        {
+            return groupBy;
+        }
+
+        public boolean presentInGroupBy(GroupingElement groupingElement)
+        {
+            for (Expression expression : groupingElement.getExpressions()) {
+                if (!presentInGroupBy(expression)) {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        public boolean presentInGroupBy(Expression expression)
```

Comment:
I would have preferred to build this as a rule validation logic. May be something for later to consider. 

Lets call it validateExpressionForGroupBy

Reply:
Lets leave a TODO as well to replace this logic with rule validation framework.